### PR TITLE
Integration tests framework part 1

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added mock Elrond/Ethereum clients
- added framework for defining "contracts" that will perform mocked functions to be used in different scenarios
- fixed a bug that could cause the relayers to panic

TODO: 
- add more functionality on the erd/eth clients, make them work as expected with the mocked contracts 
- finish first test that will prove the successfully processing of the eth->erd transactions flow
- finish test that will prove the successfully processing of the erd->eth transactions flow
- add more scenarios, use more relayers.

### CODE REVIEW

**Review by dragos-rebegea — APPROVED** [review]
:+1:

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: bridge/elrond/client.go [hunk 1]
```diff
@@ -461,6 +462,10 @@ func (c *client) executeUintQuery(valueRequest *data.VmValueRequest) (uint64, er
 		return 0, err
 	}
 
+	if len(responseData) == 0 {
+		return 0, errors.New("contract error, unexpected 0 length response data")
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply bridge/elrond/client.go:466]: done

File: integrationTests/mock/accountsMap.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+// GetAccount returns a stored account based on the provided address
+func (am *accountsMap) GetAccount(address string) *api.AccountResponse {
+	am.mutAccounts.Lock()
+	defer am.mutAccounts.Unlock()
+
+	account, exists := am.accounts[address]
+	if !exists {
+		account = &api.AccountResponse{
+			Address:         address,
+			Nonce:           0,
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply integrationTests/mock/accountsMap.go:31]: removed the default initiated fields. Kept only the `address` field.

File: integrationTests/mock/accountsMap.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+			OwnerAddress:    "",
+		}
+
+		am.accounts[address] = account
+	}
+
+	return account
+}
+
+// SetAccount will set the account & contract with the provided value
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply integrationTests/mock/accountsMap.go:48]: changed `value` to `values`. Changed to use the address from the provided objects

File: integrationTests/mock/accountsMap.go [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	defer am.mutAccounts.Unlock()
+
+	am.accounts[address] = account
+	am.contracts[address] = contract
+}
+
+// GetContract will return the contract (if existing)
+func (am *accountsMap) GetContract(address string) (*Contract, bool) {
+	am.mutAccounts.Lock()
+	defer am.mutAccounts.Unlock()
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply integrationTests/mock/accountsMap.go:60]: done

File: integrationTests/mock/contracts/elrond.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+import (
+	"fmt"
+	"math/big"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-eth-bridge/integrationTests/mock"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+)
+
+var log = logger.GetOrCreate("integrationTests/mock/constracts")
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply integrationTests/mock/contracts/elrond.go:12]: fixed

File: integrationTests/testInitializer.go [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+	Seeder           p2p.Messenger
+	Relayers         []*relay.Relay //TODO: replace this with an interface
+	ElrondClient     *mock.ElrondMockClient
+	ElrondContract   *contracts.ElrondContract
+	EthereumClient   *mock.EthereumMockClient
+	EthereumContract *mock.Contract
+	cancelFunc       func()
+}
+
+// Close will close any relayers/clients opened
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply integrationTests/testInitializer.go:36]: done

**Review by bogdan-rosianu — APPROVED** [review]

**Review by mihaieremia — APPROVED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Review by mihaieremia — APPROVED** [review]

**Pull request merged**